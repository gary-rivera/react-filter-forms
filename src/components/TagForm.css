.tag-form {
  font-size: 1rem;
  height: 2rem;
  width: 8rem;
  line-height: 80%;
  margin: 0;
}
