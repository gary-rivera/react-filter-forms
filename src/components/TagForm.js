import './TagForm.css'
import { useState } from 'react';

/** Form for add a tag for a student.
 *
 * Props:
 * - addTag (function to update StudentCard state for filters)
 *
 * { StudentCard } -> TagForm
 */

function TagForm({ addTag }) {
  const [formInput, setFormInput] = useState('');

  function handleChange(evt){
    evt.preventDefault();
    const { value } = evt.target;

    setFormInput(value)
  }

  function handleSubmit(evt){
    evt.preventDefault();
    addTag(formInput)
    setFormInput('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        className="tag-form" 
        type="text"
        placeholder="Add a tag"
        name="addTag"
        value={formInput}
        onChange={handleChange}
        autoComplete="off"
      />
    </form>
  )
}

export default TagForm;