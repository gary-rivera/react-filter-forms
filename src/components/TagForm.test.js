import { render, fireEvent } from '@testing-library/react';
import TagForm from './TagForm';

function addTagTest() {
  return;
}

it('renders without crashing', () => {
  render(<TagForm addTag={addTagTest} />);
});

it("renders the correct amount of input elements", () => {
  const { container } = render(<TagForm addTag={addTagTest} />);
  const inputs = container.querySelectorAll('input')
  expect(inputs.length).toEqual(1);
});

it("recognizes a change to input through onChange event", () => {
  const { container } = render(<TagForm addTag={addTagTest} />);
  const input = container.querySelector('input')
  fireEvent.change(input , { target:{ value: 'test'}})

  expect(input).toContainHTML('test');
});

it("matches snapshot", () => {
  const { container } = render(<TagForm addTag={addTagTest} />);
  expect(container).toMatchSnapshot();
});

