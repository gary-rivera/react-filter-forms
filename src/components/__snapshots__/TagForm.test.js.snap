// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form>
    <input
      autocomplete="off"
      class="tag-form"
      name="addTag"
      placeholder="Add a tag"
      type="text"
      value=""
    />
  </form>
</div>
`;
